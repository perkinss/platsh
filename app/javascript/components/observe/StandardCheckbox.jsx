import React from 'react'
import Checkbox from "@material-ui/core/Checkbox";

export default function StandardCheckbox(props) {

    const { id, studentId, taskId, isChecked, dispatchFormUpdate, classes } = props

    const handleChange = (event, value) => {
        let mark = { studentId: studentId, taskId: taskId, standardId: Number(id), met: value, score: value ? 1 : null }
        dispatchFormUpdate({type: 'TOGGLE_STANDARD_OBSERVATION', mark: mark})
    }

    return <Checkbox
        checked={isChecked}
        tabIndex={-1}
        onChange={handleChange}
        value={id}
        className={classes.checkbox}
    />
}
