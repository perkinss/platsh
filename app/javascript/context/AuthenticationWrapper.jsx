import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
