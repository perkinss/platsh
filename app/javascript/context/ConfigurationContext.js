import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
