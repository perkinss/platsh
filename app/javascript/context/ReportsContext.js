import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
