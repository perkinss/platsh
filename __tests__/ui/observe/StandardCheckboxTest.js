import React from 'react'
import StandardCheckbox from '../../../app/javascript/components/observe/StandardCheckbox'
import { createRender, createMount } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import { ConfigurationContext } from "../../../app/javascript/context/ConfigurationContext";

Enzyme.configure({ adapter: new Adapter() });

describe('<StandardCheckbox />', () => {
    let render, mount;
    const dispatch = () => {}
    const configuration = {
        student_marks: {
            1: {1: {1: 1}}
        }
    }
    const classes = {}

    beforeAll(() => {
        render = createRender();
        mount = createMount();
    })

    afterAll(() => {
        mount.cleanUp()
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ConfigurationContext.Provider value={{dispatch: dispatch, state: configuration}} >
                <StandardCheckbox studentId={1} taskId={1} classes={classes} id={2} />
            </ConfigurationContext.Provider>);
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

    it('should be checked if the standard is present', () => {
        const wrapper = mount(
            <ConfigurationContext.Provider value={{dispatch: dispatch, state: configuration}} >
                <StandardCheckbox studentId={1} taskId={1} classes={classes} id={1} isChecked={true} />
            </ConfigurationContext.Provider>);
        expect(wrapper).not.toBeNull()
        expect(wrapper.find('.checkbox')).not.toBeNull()
        expect(wrapper.html()).toContain('checked')
    });

    it('should not be checked if the standard is not present', () => {
        const wrapper = mount(
            <ConfigurationContext.Provider value={{dispatch: dispatch, state: configuration}} >
                <StandardCheckbox studentId={1} taskId={1} classes={classes} id={20} />
            </ConfigurationContext.Provider>);
        expect(wrapper).not.toBeNull()
        expect(wrapper.find('.checkbox')).not.toBeNull()
        expect(wrapper.html()).not.toContain('checked')
    });

})
