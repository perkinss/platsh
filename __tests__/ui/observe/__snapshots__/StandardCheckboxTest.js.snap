// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StandardCheckbox /> should match the snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "aria-disabled": "false",
      "class": "MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "MuiIconButton-label",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "PrivateSwitchBase-input-4",
              "data-indeterminate": "false",
              "tabindex": "-1",
              "type": "checkbox",
              "value": "2",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "MuiSvgIcon-root",
                "focusable": "false",
                "viewBox": "0 0 24 24",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "d": "M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                  },
                },
              ],
              "name": "svg",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
                "focusable": undefined,
                "viewBox": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
                "focusable": undefined,
                "viewBox": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-indeterminate": undefined,
              "tabindex": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-indeterminate": undefined,
              "tabindex": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
          Node {
            "attribs": Object {
              "aria-hidden": "true",
              "class": "MuiSvgIcon-root",
              "focusable": "false",
              "viewBox": "0 0 24 24",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "d": "M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "PrivateSwitchBase-input-4",
                "data-indeterminate": "false",
                "tabindex": "-1",
                "type": "checkbox",
                "value": "2",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-indeterminate": undefined,
                "tabindex": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-indeterminate": undefined,
                "tabindex": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "focusable": undefined,
              "viewBox": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "focusable": undefined,
              "viewBox": undefined,
            },
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-disabled": undefined,
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-disabled": undefined,
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
